src/utils/props.js

```diff
@@ -91,10 +91,17 @@ export function applyDefaultProps(instance, oldProps, newProps) {
 
   // update event handlers
   if (!newProps.ignoreEvents) {
-    eventHandlers.forEach(function(evt) {
-      isFunction(oldProps[evt], instance.removeListener) && instance.removeListener(evt, oldProps[evt])
-      isFunction(newProps[evt], instance.on) && instance.on(evt, newProps[evt])
-    })
+    for (let i = 0; i < eventHandlers.length; i++) {
+      const evt = eventHandlers[i]
+      if (oldProps[evt] !== newProps[evt]) {
+        if (typeof oldProps[evt] === 'function' && typeof instance.removeListener === 'function') {
+          instance.removeListener(evt, oldProps[evt])
+        }
+        if (typeof newProps[evt] === 'function' && typeof instance.on === 'function') {
```

Comment:
Do this check once outside of the loop and cache the result: `typeof instance.on === 'function'`
```
let hasOn = typeof instance.on === 'function'
for (let i = 0; i < eventHandlers.length; i++) {
   ... 
    if (typeof newProps[evt] === 'function' &&hasOn) { ... }
```